src/slash-commands/custom-voice-channels/vocal.js

```diff
@@ -0,0 +1,156 @@
+		if (!voiceChannel)
+			return interaction.reply({
+				content: 'Tu dois être dans un salon vocal pour utiliser cette commande 😕',
+				ephemeral: true,
+			})
+
+		// Si l'utilisateur n'est pas dans un salon vocal personnalisé
+		if (!client.voiceManager.has(voiceChannel.id))
+			return interaction.reply({
+				content:
+					'Tu dois être dans un salon vocal personnalisé pour utiliser cette commande 😕',
+				ephemeral: true,
+			})
+
+		switch (interaction.options.getSubcommand()) {
+			case 'no-mic':
+				// Check s'il y a déjà un salon no-mic
+				const existingNoMicChannel = client.voiceManager.get(voiceChannel.id)
+				if (existingNoMicChannel)
+					return interaction.reply({
+						content: `Il y a déjà un salon no-mic : ${existingNoMicChannel} 😕`,
+						ephemeral: true,
+					})
+
+				// Crée le salon no-mic
+				const noMicChannel = await interaction.guild.channels.create(
+					`No-mic ${voiceChannel.name}`,
+					{
+						type: Constants.ChannelTypes.GUILD_TEXT,
+						topic: `Salon temporaire créé pour ${displayNameAndID(
+							interaction.member,
+							interaction.user,
+						)}`,
+						parent: voiceChannel.parent,
+						// Setup les permissions (pas d'accès)
+						// pour le rôle everyone
+						permissionOverwrites: [
+							{
+								id: interaction.guild.id,
+								deny: [
+									Permissions.FLAGS.CREATE_INSTANT_INVITE,
+									Permissions.FLAGS.MANAGE_CHANNELS,
+									Permissions.FLAGS.MANAGE_ROLES,
+									Permissions.FLAGS.MANAGE_WEBHOOKS,
+									Permissions.FLAGS.VIEW_CHANNEL,
+									Permissions.FLAGS.SEND_MESSAGES,
+									Permissions.FLAGS.SEND_TTS_MESSAGES,
+									Permissions.FLAGS.MANAGE_MESSAGES,
+									Permissions.FLAGS.EMBED_LINKS,
+									Permissions.FLAGS.ATTACH_FILES,
+									Permissions.FLAGS.READ_MESSAGE_HISTORY,
+									Permissions.FLAGS.MENTION_EVERYONE,
+									Permissions.FLAGS.USE_EXTERNAL_EMOJIS,
+									Permissions.FLAGS.USE_EXTERNAL_STICKERS,
+									Permissions.FLAGS.ADD_REACTIONS,
+								],
+							},
+						],
+					},
+				)
+
+				// Setup des permissions
+				await Promise.all(
+					// Accès au salon pour les membres présents
+					voiceChannel.members.map(member => {
+						// eslint-disable-next-line no-unused-expressions
+						member.permissions.has(Permissions.FLAGS.MANAGE_MESSAGES)
+							? noMicChannel.permissionOverwrites.edit(member.id, {
+									VIEW_CHANNEL: true,
+									SEND_MESSAGES: true,
+									READ_MESSAGE_HISTORY: true,
+									MANAGE_MESSAGES: true,
+							  })
+							: noMicChannel.permissionOverwrites.edit(member.id, {
+									VIEW_CHANNEL: true,
+									SEND_MESSAGES: true,
+									READ_MESSAGE_HISTORY: true,
+							  })
+					}),
+				)
+
+				// Ajout du salon dans la map
+				client.voiceManager.set(voiceChannel.id, noMicChannel)
+
+				return interaction
+					.reply({
+						content: `Ton salon a bien été créé : ${noMicChannel} 👌`,
+						ephemeral: true,
+					})
+					.catch(() => null)
```

Comment:
Quelle(s) erreur(s) peu(ven)t être(s) générée(s) ? Il faudrait les gérer proprement plutôt que de toujours renvoyer null

Reply:
Le null a été mis car parfois si la création prend du temps, l'interaction répond "The application did not respond", mais ne génère aucune erreur
ça permettait de ne pas afficher cette réponse

Reply:
Si parfois c'est trop long, on peut utiliser `interaction.deferReply()` pour étendre la validié du token et éviter ça : https://discordjs.guide/interactions/replying-to-slash-commands.html#deferred-responses

Reply:
https://github.com/TanguyChiffoleau/Le-bot-en-JS/pull/258/commits/a9ecd101a2b8a81a622dbd67a9f9d56b648e705f